routing/pathfind.go

```diff
@@ -305,6 +309,50 @@ type PathFindingConfig struct {
 	MinProbability float64
 }
 
+// getMaxOutgoingAmt returns the maximum available balance in any of the
+// channels of the given node.
+func getMaxOutgoingAmt(node route.Vertex, outgoingChan *uint64,
+	g *graphParams, tx *bbolt.Tx) (lnwire.MilliSatoshi, error) {
+
+	var max lnwire.MilliSatoshi
+	cb := func(_ *bbolt.Tx, edgeInfo *channeldb.ChannelEdgeInfo, outEdge,
+		_ *channeldb.ChannelEdgePolicy) error {
+
+		if outEdge == nil {
+			return nil
+		}
+
+		chanID := outEdge.ChannelID
+
+		// Enforce outgoing channel restriction.
+		if outgoingChan != nil && chanID != *outgoingChan {
+			return nil
+		}
+
+		bandwidth, ok := g.bandwidthHints[chanID]
+
+		// If the bandwidth is not available for whatever reason, don't
+		// fail the pathfinding early.
+		if !ok {
+			max = lnwire.MaxMilliSatoshi
```

Comment:
If there is no bandwidth hint for just one of the channels, this causes the `getMaxOutgoingAmt` to always return `lnwire.MaxMilliSatoshi` because no other channel can overwrite the max with a "real" value. Should this maybe just be ignored? I guess it depends on what the causes for the hints being empty could be and if it's likely that the hint for just one channel is missing or that the hints for all of them are missing.

Reply:
There may be a race condition. A channel becoming available right after the bandwidth hints are queried. It is an edge case and it felt safer to do it like this. Not to fail the routing while there is a channel available that just doesn't have a bandwidth hint.